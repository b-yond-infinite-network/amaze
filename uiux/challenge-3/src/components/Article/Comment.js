import DeleteButton from './DeleteButton';
import { Link } from 'react-router-dom';
import React from 'react';

const Comment = props => {
  const comment = props.comment;
  const show = props.currentUser &&
    props.currentUser.username === comment.author.username;
  return (
    <div id="comment-card" className="card">
      <div className="card-block">
        <p className="card-text">{comment.body}</p>
      </div>
      <div className="card-footer">
        <Link
        id="comment-author-image"
          to={`/@${comment.author.username}`}
          className="comment-author">
          <img src={comment.author.image} className="comment-author-img" alt={comment.author.username} />
        </Link>
        &nbsp;
        <Link
        id="comment-author"
          to={`/@${comment.author.username}`}
          className="comment-author">
          {comment.author.username}
        </Link>
        <span id="comment-created" className="date-posted">
          {new Date(comment.createdAt).toDateString()}
        </span>
        <DeleteButton id="comment-delete-button" show={show} slug={props.slug} commentId={comment.id} />
      </div>
    </div>
  );
};

export default Comment;
