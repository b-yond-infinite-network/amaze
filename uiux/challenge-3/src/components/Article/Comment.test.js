import Comment from './Comment';
import { shallow } from 'enzyme';
import React from 'react';
import renderer from 'react-test-renderer'
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router'
import configureMockStore from 'redux-mock-store'
import middlewares from './../../store'

const initialState = {
    currentUser: {
        username: "test username"
    },
    comment: {
        id: '1234',
        body: 'Test body',
        author: {
            username: 'test username',
            image: ''
        },
        createdAt: '2019-06-29T18:36:23.216Z'
    }
};

describe('Comment Component', () => {

    let mockStore;
    let comment;
    let currentUser;

    let tree;
    let wrapper;
    let render;

    describe('currentUser === author', () => {

        beforeAll(() => {

            mockStore = configureMockStore(middlewares)({ initialState });
            comment = initialState.comment;
            currentUser = initialState.currentUser;

            tree = <Provider store={mockStore}>
                <MemoryRouter initialEntries={["/"]}>
                    <Comment comment={comment} currentUser={currentUser} />
                </MemoryRouter>
            </Provider>;
            wrapper = shallow(tree).childAt(0).childAt(0).dive();
            console.log(wrapper.debug());
            render = renderer.create(tree).toJSON();

        });

        it('Should match the snapshot', () => {
            expect(render).toMatchSnapshot();
        });

        it('Should render without errors', () => {
            const component = wrapper.find('#comment-card');
            expect(component.length).toBe(1);
        });

        it('Should render author image', () => {
            const component = wrapper.find('#comment-author-image');
            expect(component.length).toBe(1);
        });

        it('Should render author info', () => {
            const component = wrapper.find('#comment-author');
            expect(component.length).toBe(1);
        });

        it('Should render delete button', () => {
            const component = wrapper.find('#comment-delete-button');
            expect(component.length).toBe(1);
        });

        it('Should show delete button', () => {
            const component = wrapper.find('#comment-delete-button');
            expect(component.props()['show']).toEqual(true);
        });

    });

    describe('currentUser !== author', () => {

        beforeAll(() => {

            mockStore = configureMockStore(middlewares)({ initialState });
            comment = initialState.comment;
            currentUser = {};

            tree = <Provider store={mockStore}>
                <MemoryRouter initialEntries={["/"]}>
                    <Comment comment={comment} currentUser={currentUser} />
                </MemoryRouter>
            </Provider>;
            wrapper = shallow(tree).childAt(0).childAt(0).dive();
            console.log(wrapper.debug());
            render = renderer.create(tree).toJSON();

        });

        it('Should match the snapshot', () => {
            expect(render).toMatchSnapshot();
        });

        it('Should render without errors', () => {
            const component = wrapper.find('#comment-card');
            expect(component.length).toBe(1);
        });

        it('Should render author image', () => {
            const component = wrapper.find('#comment-author-image');
            expect(component.length).toBe(1);
        });

        it('Should render author info', () => {
            const component = wrapper.find('#comment-author');
            expect(component.length).toBe(1);
        });

        it('Should render delete button', () => {
            const component = wrapper.find('#comment-delete-button');
            expect(component.length).toBe(1);
        });

        it('Should NOT show delete button', () => {
            const component = wrapper.find('#comment-delete-button');
            expect(component.props()['show']).toEqual(false);
        });

    });

});