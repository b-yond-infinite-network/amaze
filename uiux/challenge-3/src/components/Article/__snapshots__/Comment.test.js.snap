// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Comment Component currentUser !== author Should match the snapshot 1`] = `
<div
  className="card"
  id="comment-card"
>
  <div
    className="card-block"
  >
    <p
      className="card-text"
    >
      Test body
    </p>
  </div>
  <div
    className="card-footer"
  >
    <a
      className="comment-author"
      href="/@test username"
      id="comment-author-image"
      onClick={[Function]}
    >
      <img
        alt="test username"
        className="comment-author-img"
        src=""
      />
    </a>
     
    <a
      className="comment-author"
      href="/@test username"
      id="comment-author"
      onClick={[Function]}
    >
      test username
    </a>
    <span
      className="date-posted"
      id="comment-created"
    >
      Sat Jun 29 2019
    </span>
  </div>
</div>
`;

exports[`Comment Component currentUser === author Should match the snapshot 1`] = `
<div
  className="card"
  id="comment-card"
>
  <div
    className="card-block"
  >
    <p
      className="card-text"
    >
      Test body
    </p>
  </div>
  <div
    className="card-footer"
  >
    <a
      className="comment-author"
      href="/@test username"
      id="comment-author-image"
      onClick={[Function]}
    >
      <img
        alt="test username"
        className="comment-author-img"
        src=""
      />
    </a>
     
    <a
      className="comment-author"
      href="/@test username"
      id="comment-author"
      onClick={[Function]}
    >
      test username
    </a>
    <span
      className="date-posted"
      id="comment-created"
    >
      Sat Jun 29 2019
    </span>
    <span
      className="mod-options"
      id="delete-button"
    >
      <i
        className="ion-trash-a"
        onClick={[Function]}
      />
    </span>
  </div>
</div>
`;
